CoffeeShop.cpp

```diff
@@ -0,0 +1,163 @@
+#include <iostream>
+#include "CoffeeShop.h"
+
+CoffeeShop::CoffeeShop() :
+numBaristas(2),
+numCoffeeMachines(4),
+numRegisters(2),
+numCustomers(0),
+numTables(12),
+balance(0.0f),
+brewMethod("pour over")
+{
+    std::cout << "\n+ CoffeeShop constructed!" << std::endl;  
+}
+
+CoffeeShop::~CoffeeShop()
+{
+    std::cout << "\n- CoffeeShop destructed!" << std::endl;
+}
+
+CoffeeShop::Coffee::Coffee() :
+type("columbian"),
+roast("dark"),
+coarseness(1)
+{
+    std::cout << "\n+ CoffeeShop::Coffee constructed!" << std::endl;  
+}
+
+CoffeeShop::Coffee::~Coffee()
+{
+    std::cout << "\n- CoffeeShop::Coffee destructed!" << std::endl;  
+}
+
+CoffeeShop::Customer::Customer() :
+customerID(1),
+rewardsMember(false),
+customerPhoneNumber(0.0f),
+rewardsBalance(0.0f),
+name("New Customer")
+{
+    std::cout << "\n+ CoffeeShop::Customer constructed!" << std::endl;  
+
+}
+
+CoffeeShop::Customer::~Customer()
+{
+    std::cout << "\n- CoffeeShop::Customer destructed!" << std::endl;  
+
+}
+
+bool CoffeeShop::Customer::contactCustomer(std::string msg)
+{
+    if( customerPhoneNumber > 0 )
+    {
+        if( msg.length() > 1 )
+        {
+            std::cout << "Texting " << name << ": \"" << msg << "\"" << std::endl;
+            return true;
+        }
+        else std::cout << "Message is too short.\n";
+
+    }
+    else std::cout << name << " doesn't have a phone number saved.\n";
+
+    return false;
+}
+
+bool CoffeeShop::Customer::useRewardsPoints(float rewardsPoints)
+{
+    if( rewardsMember == true )
+    {
+        if( rewardsBalance > rewardsPoints )
+        {
+            rewardsBalance -= rewardsPoints;
+            std::cout << name << " used " << rewardsPoints << " points, leaving a balance of " << rewardsBalance << "\n";
+            return true;
+        }
+        else std::cout << name << " doesn't have enough points.\n";
+
+    }
+    else std::cout << name << " is not a rewards member!\n";
```

Comment:
If you use curly braces in the `if` block, use them with the `else` block
- [ ] revise